Adding Dialog component

loweisz opened the pull request:
Based on #38 

### Implementation

Added a simple Dialog component. 💬 
As suggested I installed and mostly used [@reach/dialog](https://reacttraining.com/reach-ui/dialog/#dialog) as a basis. 

For the general `Dialog` component I couldn't use the `Dialog` component from @reach/dialog, because the custom styling wasn't applied there. 
So I built it with `DialogOverlay` and `DialogContent`, which works fine with the custom styles. 
But open for suggestions for better approaches 😃 

### Documentation

I added all three components to the documentation, but with a very simple basic explanation, mostly linking to [@reach/dialog](https://reacttraining.com/reach-ui/dialog/#dialog), since the api is almost completely based on it.

rubenmoya commented on packages/react-ui/src/components/dialog/index.js:
```diff
@@ -0,0 +1,46 @@
+    as={ReachDialog.DialogOverlay}
+    component="DialogOverlay"
+    css={merge(styles.DialogOverlay, css)}
+    {...props}
```
Do we want to add the same `props` to both elements? That could lead to problems if it adds duplicated props, imagine a `data-attribute` or an `id`

loweisz replied:
Yes makes sense, I removed the `props` from the overlay and have it now only on the `DialogContent` component, since this is the "core" component, I think. Maybe it would make sense to add a `props` called `overlayProps` to add props to the overlay, what do you think?

siddharthkp commented:
Hey! 

This is really strong start! Also, thanks for putting up with the totally random docs structure 😅 


👍 
Love that it's already accessible!
Love that we have docs!
I think the default styles are very reasonable!
Theme components names look good!

🤔 
I don't think we need to export the same abstractions as reach-ui. I'd imagine we would be fine with just `Dialog` as long as you can style overlay and content via their names (similar to the implementation in [Breadcrumb](https://react-ui.dev/components/Breadcrumb))

loweisz commented:
Thank you @siddharthkp 😄 
I totally agree on having only the dialog component and not the splitted ones. I wanted to do it as close as possible to the reach-ui implementation, but the simplicity is more important here I think 👍 

I removed both the components `DialogContent` and `DialogOverlay` from the docs and from the library. 
I also extracted the `isOpen` and `onDismiss` from the props and pass them only to the overlay, since this is the only one wanting it. The rest of the props, as suggested from @rubenmoya (Thank you 😄 ) are only spreaded to the `DialogContent` component since I think this is the key component here.

rubenmoya commented on packages/docs/src/pages/components/dialog.js:
```diff
@@ -0,0 +1,186 @@
+  const [showDialog, setShowDialog] = React.useState(false)
+  const open = () => setShowDialog(true)
+  const close = () => setShowDialog(false)
+  console.log(showDialog)
```
I think we can remove this 🔥

loweisz replied:
Oops, should be gone 👍

siddharthkp commented:
Hey! Thanks for taking the time! Merged this into master. Will publish in the next version this week

FYI: I'm tweaking the styles just a bit to match my design style 🙃 
https://github.com/siddharthkp/react-ui/pull/64

loweisz commented:
Hey, Very cool ! Thank you! Was happy to help you a bit, especially because I really like the idea behind it! 
I'm open to new contributions, if there anything I could help you with 😃
